controllers/cluster_controller_test.go

```diff
@@ -49,4 +50,37 @@ var _ = Describe("Cluster Reconciler", func() {
 		}, timeout).Should(BeTrue())
 	})
 
+	It("Should patch the status even if the spec is empty", func() {
```

Comment:
Could we also add some tests around `patchCluster` to verify the edge cases there?

Reply:
Ah, that message needs improvement.

It does both update status if spec is empty and update spec if status is empty.

Reply:
If that's not satisfactory can you please tell me which edge cases you'd like tests for?

Reply:
I'd split it in 2 parts :)

Reply:
haha i just went the opposite route and made the test name very generic. I can do two different tests if you like though

Reply:
I was thinking of the following three scenarios:
- Only status updated 
- Only non-status updated
- Both updated
Tested independently to ensure we see the expected changes reflected in the resulting object.